app/src/main/java/com/celzero/bravedns/util/Utilities.kt

```diff
@@ -250,6 +260,36 @@ class Utilities {
             } else String.format(Locale.ROOT, "%d", type)
         }
 
+        fun getThemeAccent(context: Context): Int {
+            return if (isDarkSystemTheme(context)) {
+                R.color.accentGoodBlack
+            } else {
+                R.color.negative_white
+            }
+        }
+
+        private fun isDarkSystemTheme(context: Context): Boolean {
+            return context.resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK == Configuration.UI_MODE_NIGHT_YES
+        }
+
+        fun openVPNProfile(context: Context) {
+            try {
+                val intent = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                    Intent(ACTION_VPN_SETTINGS)
+                } else {
+                    Intent("android.net.vpn.SETTINGS")
+                }
+                intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
+                context.startActivity(intent)
+            } catch (e: ActivityNotFoundException) {
+                showToastInMidLayout(context, context.getString(R.string.vpn_profile_error), Toast.LENGTH_SHORT)
+                Log.w(LOG_TAG, "Exception while opening app info: ${e.message}", e)
+            }
+        }
+
+        fun isValidUid(uid: Int): Boolean {
+            return uid != INVALID_UID || uid != MISSING_UID
```

Comment:
shouldn't this be `return uid >= 0 && uid < ...`?

ref: https://cs.android.com/android/platform/superproject/+/master:system/core/libcutils/include/private/android_filesystem_config.h?q=android_filesystem_config.h

Reply:
Modified to check the range.